common/src/main/java/com/microsoft/identity/common/internal/request/SdkType.java

```diff
@@ -22,10 +22,39 @@
 // THE SOFTWARE.
 package com.microsoft.identity.common.internal.request;
 
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+
 /**
  * Enum to indicate if the request came from an app hosting Adal or Msal Sdk
  */
 public enum SdkType {
     ADAL,
-    MSAL
-}
\ No newline at end of file
+    MSAL,
+    MSAL_CPP,
+    UNKNOWN;
+
+    /**
+     * Method for mapping the SdkType to appropriate String
+     * for the purpose of sending it to the telemetry.
+     */
+    public String getProductName() {
+        if ((SdkType.ADAL == this) || (SdkType.MSAL == this)) {
+            return AuthenticationConstants.SdkPlatformFields.PRODUCT_NAME_MSAL;
+        } else if ((SdkType.MSAL_CPP == this)) {
+            return AuthenticationConstants.SdkPlatformFields.PRODUCT_NAME_MSAL_CPP;
+        } else {
+            // value is intended for test-cases, eg. CommandDispatcherTest.java.
+            return "";
```

Comment:
Aren't we using default of `unknown` in this case?

Reply:
yes, we are using it in CommandDispatcher.java for submitSilent and beginInteractive methods. this field is primarily used to adhere to already written test-cases.